content/documentation/pages/2-concepts/1-architecture.md

```diff
@@ -1,33 +1,330 @@
 ---
 path: 'concepts/architecture/'
 title: 'Architecture'
-description: 'Lorem markdownum madefacta, circumtulit aliis, restabat'
+description: "Introduction to Data Flow's Architecture."
 ---
 
 # Architecture
 
-TODO
+Thi guide explains the main concepts of Data Flow's architecture:
 
-## System Requirements
+- Data Flow's Server Components
+- The types of applications the server components can deploy for Streams and Batch Jobs
+- The microservice architecture of the deployed applications and how that is defined using a DSL
+- The platforms where they are deployed
 
-**Java:** Data Flow uses Java 8.
+Other sections of the guide explain
 
-**Database:** The Data Flow Server and Skipper Server need to have an RDBMS installed.
+- How the Server Components are secured and the tooling used to interact with them.
+- Runtime monitoring of Stream data pipelines.
+- The Spring projects that you can use to develop Stream and Batch data pipelines
+
+## Server Components
+
+Data Flow has two main components:
+
+- Data Flow Server
+- Skipper Server
+
+The main entry point to access Data Flow is through the Web API of the Data Flow Server.
+The Web Dashboard is served from the Data Flow Server and a separate Data Flow Shell applications both communicate through the web API.
+
+The servers can be run on several platforms: Cloud Foundry, Kubernetes, or on your Local machine.
+Each server stores state in a relational database.
```

Comment:
The term "state" makes it sound something big like other monolith runtimes do. Maybe: "Each server stores lightweight metadata about the streaming and batch data pipelines in a relational database".

Reply:
well there is also the batch schema, skipper state machine etc.  so it isn't just 2-3 tables....